Bluetooth: Mesh: no more tinycrypt in ble mesh tfm images

alxelax opened the pull request:
PR allows to get rid of tinycrypt objects from
the final binary of the ble mesh apps based on
PSA TFM crypto.

Final size of mesh app for nrf5340dk_nrf5340_cpuapp_ns (that is build with TFM PSA crypto)
```
Memory region         Used Size  Region Size  %age Used
           FLASH:      120936 B       192 KB     61.51%
             RAM:       35350 B       192 KB     17.98%
        IDT_LIST:          0 GB         2 KB      0.00%
```

without these changes tinycrypt sneaks into it
```
Memory region         Used Size  Region Size  %age Used
           FLASH:      126120 B       192 KB     64.15%
             RAM:       37294 B       192 KB     18.97%
        IDT_LIST:          0 GB         2 KB      0.00%
```
+
some bsim tests fixes due to collisions after timing changes.

alxelax committed:
Bluetooth: Mesh: no more tinycrypt in ble mesh tfm image

PR allows to get rid of tinycrypt objects from
the final binary of the ble mesh apps based on
PSA TFM crypto.

Signed-off-by: Aleksandr Khromykh <aleksandr.khromykh@nordicsemi.no>

akredalen commented on subsys/bluetooth/host/Kconfig:
```diff
@@ -163,12 +163,15 @@ rsource "../mesh/Kconfig"
+	help
+	  The option adds the AES encryption support using TinyCrypt
+	  library if this is not provided by the controller implementation.
+	  Either the option is not required if BLE Mesh is enabled and
```
```suggestion
	  The option is not required if BLE Mesh is enabled and
```

akredalen commented on samples/bluetooth/mesh/boards/nrf5340dk_nrf5340_cpuapp_ns.conf:
```diff
@@ -1,3 +1,8 @@
+# The option adds tinycrypt based bt_rand.
```
```suggestion
# The option adds TinyCrypt based bt_rand.
```
Same for three other .conf-files.

akredalen commented on samples/bluetooth/mesh/boards/nrf5340dk_nrf5340_cpuapp_ns.conf:
```diff
@@ -1,3 +1,8 @@
+# The option adds tinycrypt based bt_rand.
+CONFIG_BT_HOST_CRYPTO=n
+# The option adds gatt caching feature that is based on tinycrypt.
```
```suggestion
# The option adds GATT caching feature that is based on TinyCrypt.
```
Same for three other .conf-files.

alxelax committed:
samples: Bluetooth: Mesh: add main stack for mesh provisioner

The current main stack size is not enough for any platform.
This commit stack size was checked for nrf52840 and nrf5340.
Everything works.

Signed-off-by: Aleksandr Khromykh <aleksandr.khromykh@nordicsemi.no>

PavelVPV commented on subsys/bluetooth/host/Kconfig:
```diff
@@ -163,12 +163,15 @@ rsource "../mesh/Kconfig"
+	help
+	  The option adds the AES encryption support using TinyCrypt
+	  library if this is not provided by the controller implementation.
+	  The option is not required if BLE Mesh is enabled and
```
Not sure if mesh should be mentioned here. What about `bt_encrypt_le` and `bt_encrypt_be` functions? They aren't replaced when mesh is enabled with PSA support. This options still needs to be enabled if a customer wants to enable SMP (Security Manager Protocol, not SMP from MCUmgr).

alxelax replied:
Mesh doesn't use them. Yes, if customer adds anything that uses them, tinycrypt will appear again. I guess there is no common strategy to get rid tinycrypt in Zephyr in general.

PavelVPV commented on samples/bluetooth/mesh_provisioner/boards/nrf5340dk_nrf5340_cpuapp_ns.conf:
```diff
@@ -1,3 +1,8 @@
+# The option adds TinyCrypt based bt_rand.
+CONFIG_BT_HOST_CRYPTO=n
+# The option adds GATT caching feature that is based on TinyCrypt.
+CONFIG_BT_GATT_CACHING=n
```
Is `CONFIG_BT_GATT_CACHING=n` needed here? mesh_provisioner doesn't enable neither peripheral nor central role support.

alxelax commented on samples/bluetooth/mesh/boards/nrf5340dk_nrf5340_cpuapp_ns.conf:
```diff
@@ -1,3 +1,8 @@
+# The option adds TinyCrypt based bt_rand.
+CONFIG_BT_HOST_CRYPTO=n
+# The option adds GATT caching feature that is based on TinyCrypt.
+CONFIG_BT_GATT_CACHING=n
```
I guess this is not sufficient. Host also supports ecc, aes ccm and randomization with tinycrypt. Host requires significant refactoring to get rid of tinycrypt. This is quite huge work that we've already done for mesh.
